crates/rpc/rpc/src/trace.rs

```diff
@@ -164,6 +164,38 @@ where
             .ok_or_else(|| EthApiError::TransactionNotFound)?
     }
 
+    /// Replays all transaction in a block
+    pub async fn replay_block_transactions(
+        &self,
+        block_id: BlockId,
+        trace_types: HashSet<TraceType>,
+    ) -> Result<Option<Vec<TraceResultsWithTransactionHash>>> {
```

Comment:
this is basically the same thing as `trace_block`, but the recorded traces differ: 

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/trace.rs#L238-L241

what we should do here is:

Convert `trace_block` into a helper function that can also unify `replay_block_transactions`. This new function should accept a `TracingInspectorConfig` and a callback function `FnOnce()` like this:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/transactions.rs#L136-L140

so both trace block functions call this new helper functions but with a different config and callback (that converts the inspector into traces)

does this make sense?

Reply:
Okay 
Yes this make sense